packages/ndla-ui/src/AudioPlayer/Controls.tsx

```diff
@@ -0,0 +1,519 @@
+  const [speedValue, setSpeedValue] = useState(1);
+  const [volumeValue, setVolumeValue] = useState(100);
+  const [sliderValue, setSliderValue] = useState(0);
+  const [currentTime, setCurrentTime] = useState(0);
+  const [remainingTime, setRemainingTime] = useState(0);
+  const [playing, setPlaying] = useState(false);
+  const audioRef = useRef<HTMLAudioElement>(null);
+
+  useEffect(() => {
+    if (audioRef.current) {
+      audioRef.current.playbackRate = speedValue;
+    }
+  }, [speedValue]);
+
+  useEffect(() => {
+    if (audioRef.current) {
+      const audioElement = audioRef.current;
+      const handleTimeUpdate = () => {
+        const { currentTime, duration } = audioElement;
+        const percent = Math.round((currentTime / duration) * 100);
+        setSliderValue(percent);
+        setCurrentTime(Math.round(currentTime));
+        setRemainingTime(Math.round(duration - currentTime));
+      };
+
+      const handleLoadedMetaData = () => {
+        const { currentTime, duration } = audioElement;
+        setCurrentTime(Math.round(currentTime));
+        setRemainingTime(Math.round(duration - currentTime));
+        setRemainingTime(Math.round(duration - currentTime));
+      };
+
+      const handleTimeEnded = () => {
+        setPlaying(false);
+      };
+
+      audioElement.addEventListener('timeupdate', handleTimeUpdate);
+      audioElement.addEventListener('loadedmetadata', handleLoadedMetaData);
+      audioElement.addEventListener('ended', handleTimeEnded);
+      return () => {
+        audioElement.removeEventListener('timeupdate', handleTimeUpdate);
+        audioElement.removeEventListener(
+          'loadedmetadata',
+          handleLoadedMetaData,
+        );
+        audioElement.removeEventListener('ended', handleTimeEnded);
+      };
+    }
+  }, []);
+
+  useEffect(() => {
+    if (audioRef.current) {
+      const audioElement = audioRef.current;
+      if (playing) {
+        audioElement.play();
+      } else {
+        audioElement.pause();
+      }
+    }
+  }, [playing]);
+
+  const togglePlay = () => {
+    setPlaying(!playing);
+  };
+
+  const onSeekSeconds = (seconds: number) => {
+    if (audioRef.current) {
+      audioRef.current.currentTime += seconds;
+    }
+  };
+
+  const handleSliderChange = (value: number) => {
+    if (audioRef.current) {
+      audioRef.current.currentTime = (value / 100) * audioRef.current.duration;
+    }
+  };
+
+  const handleVolumeSliderChange = (value: number) => {
+    if (audioRef.current) {
+      audioRef.current.volume = value / 100;
+      setVolumeValue(value);
+    }
+  };
+
+  return (
+    <>
+      <audio ref={audioRef} src={src} title={title} preload="metadata" />
+      <ControlsWrapper>
+        <PlayButton
+          type="button"
+          onClick={togglePlay}
+          title="play"
+          aria-label="play">
+          <span aria-hidden>
+            {playing ? (
+              <Pause role="img" aria-label="pause" title="play" />
```

Comment:
Det som ikkje gir meining er at både role og aria-label er aria props, men TS reagerte kun på role. Så eg måtte legge den til på IconProps for at koden skulle validere.